### PR TITLE
CI: Change development python version to 3.10

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Mypy passes locally now. Lets see if ci passes as well.

### CODE REVIEW

**Comment by phofl** [comment]
cc @mroeschke any idea why it's not including the Series.dt file?

**Comment by mroeschke** [comment]
It looks like it's being included in https://pandas.pydata.org/preview/51133/docs/reference/api/pandas.Series.dt.date.html?

**Comment by phofl** [comment]
I meant https://pandas.pydata.org/docs/reference/api/pandas.Series.dt.html in our current docs. Sphinx raises a warning that the document isn't included in any toctree. the str and cat accessors work

**Comment by mroeschke** [comment]
Ah hmm yeah I'm not too sure off the top of my head why it's not included. None of the sphinx versions changed with the bump right?

**Comment by phofl** [comment]
No they are the same. This is really weird, the inheritance logic is a bit weird for the date accessor, so I think that's the problem. Would you be ok with merging anyway?

**Comment by mroeschke** [comment]
I guess it shouldn't necessarily block dropping 3.8, but lets definitely raise an issue about Series.dt being missing from API reference potentially

**Comment by phofl** [comment]
We should address the sphinx stuff but can do as a follow up. Have to get ci green here

**Comment by phofl** [comment]
I skipped Pypy for now. @lithomas1 any idea how to install PyPy for 3.9?

Otherwise this looks good now, but should merge the unpin matplotlib pr first.

File: ci/deps/actions-pypy-39.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ dependencies:
   # TODO: Add the rest of the dependencies in here
   # once the other plentiful failures/segfaults
   # with base pandas has been dealt with
-  - python=3.8[build=*_pypy] # TODO: use this once pypy3.8 is available
+  - python=3.9.0[build=*_pypy] # TODO: use this once pypy3.8 is available
```
Comment (lithomas1) [inline ci/deps/actions-pypy-39.yaml:8]:
```suggestion
  - python=3.9[build=*_pypy]
```
There is no PyPy version of Python 3.9.0

**Comment by phofl** [comment]
Thx, this should be good now

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ jobs:
         - [windows-2019, win_amd64]
         - [windows-2019, win32]
         # TODO: support PyPy?
-        python: [["cp38", "3.8"], ["cp39", "3.9"], ["cp310", "3.10"],  ["cp311", "3.11"]]# "pp38", "pp39"]
+        python: [["cp39", "3.9"], ["cp310", "3.10"],  ["cp311", "3.11"]]# "pp38", "pp39"]
```
Comment (mroeschke) [inline .github/workflows/wheels.yml:58]:
```suggestion
        python: [["cp39", "3.9"], ["cp310", "3.10"],  ["cp311", "3.11"]]# "pp39"]

```

File: doc/source/conf.py [hunk 3]
```diff
@@ -76,6 +76,7 @@
     # to ensure that include files (partial pages) aren't built, exclude them
     # https://github.com/sphinx-doc/sphinx/issues/1965#issuecomment-124732907
     "**/includes/**",
+    "**/api/pandas.Series.dt.rst",
```
Comment (mroeschke) [inline doc/source/conf.py:79]: Just a reminder to open an issue about this once we merge this

**Comment by mroeschke** [comment]
This would need a whatsnew in 2.1 that 3.8 support has been dropped

**Comment by mroeschke** [comment]
Some grepped results where we still mention 3.8

```
./doc/source/getting_started/install.rst:Officially Python 3.8, 3.9, 3.10 and 3.11.
./doc/source/development/contributing_gitpod.rst:We have built a python 3.8 environment and all development dependencies will
```

File: doc/source/conf.py (hunk 3 above)
Comment (phofl) [reply doc/source/conf.py:79]: Yep, it's on my todo list

**Comment by phofl** [comment]
Thx, completely forgot the docs. Updated

**Review by mroeschke — APPROVED** [review]

**Review by lithomas1 — APPROVED** [review]
LGTM. Are you going to remove ``pandas.compat.PY39`` as well here or in a followup?

**Comment by phofl** [comment]
This is getting bigger and bigger (underestimated it when starting to rip out 3.8), would prefer merging and then doing the rest as follow ups

**Comment by lithomas1** [comment]
Sounds good to me.
thanks @phofl

**Pull request merged**